sw/device/silicon_creator/lib/drivers/pinmux.c

```diff
@@ -47,32 +30,60 @@ typedef struct pinmux_output {
 } pinmux_output_t;
 
 /**
- * MIO pad to peripheral output mappings.
+ * UART RX pin.
  */
-static const pinmux_output_t kPinmuxOutputs[] = {
-    /**
-     * UART
-     */
-    {.pad = kTopEarlgreyPinmuxMioOutIoc4,
-     .periph = kTopEarlgreyPinmuxOutselUart0Tx},
+static const pinmux_input_t kInputUart = {
+    .periph = kTopEarlgreyPinmuxPeripheralInUart0Rx,
+    .pad = kTopEarlgreyPinmuxInselIoc3,
 };
 
+/**
+ * Bootstrap pin.
+ */
+static const pinmux_input_t kInputBootstrap = {
+    .periph = kTopEarlgreyPinmuxPeripheralInGpioGpio17,
+    .pad = kTopEarlgreyPinmuxInselIob8,
+};
+
+/**
+ * UART TX pin.
+ */
+static const pinmux_output_t kOutputUart = {
+    .pad = kTopEarlgreyPinmuxMioOutIoc4,
+    .periph = kTopEarlgreyPinmuxOutselUart0Tx,
+};
+
+/**
+ * Sets the input pad for the specified peripheral input.
+ *
+ * @param input A peripheral input and MIO pad to link it to.
+ */
+static void configure_input(pinmux_input_t input) {
+  abs_mmio_write32(TOP_EARLGREY_PINMUX_AON_BASE_ADDR +
+                       PINMUX_MIO_PERIPH_INSEL_0_REG_OFFSET +
+                       input.periph * sizeof(uint32_t),
+                   input.pad);
+}
+
+/**
+ * Sets the peripheral output for each specified output pad.
+ *
+ * @param output An MIO pad and a peripheral output to link it to.
+ */
+static void configure_output(pinmux_output_t output) {
+  abs_mmio_write32(TOP_EARLGREY_PINMUX_AON_BASE_ADDR +
+                       PINMUX_MIO_OUTSEL_0_REG_OFFSET +
+                       output.pad * sizeof(uint32_t),
+                   output.periph);
+}
+
 void pinmux_init(void) {
-  // Set the input pad for each specified peripheral input.
-  const uint32_t kInputBase =
-      TOP_EARLGREY_PINMUX_AON_BASE_ADDR + PINMUX_MIO_PERIPH_INSEL_0_REG_OFFSET;
-  for (uint32_t i = 0; i < ARRAYSIZE(kPinmuxInputs); ++i) {
-    uint32_t reg = kPinmuxInputs[i].periph * sizeof(uint32_t);
-    uint32_t val = kPinmuxInputs[i].pad;
-    abs_mmio_write32(kInputBase + reg, val);
+  uint32_t bootstrap_en = otp_read32(OTP_CTRL_PARAM_ROM_BOOTSTRAP_EN_OFFSET);
+  if (launder32(bootstrap_en) == kHardenedBoolTrue) {
+    HARDENED_CHECK_EQ(bootstrap_en, kHardenedBoolTrue);
+    configure_input(kInputBootstrap);
```

Comment:
Should we make configuring the pins conditional on the OTP word or should we unconditionally configure the strapping pins as inputs and use the OTP word to control entry into bootstrap mode?

ROM_EXT and owner code are probably going to want to read the straps too.

Reply:
I was planning to control both (pin configuration and bootstrap entry) using the OTP value (no need to configure the pins if we are not going to use them) but can revert this if you think we should configure the pins. @moidx any thoughts?